x/permission/types/keys.go

```diff
@@ -93,6 +100,49 @@ func GetGroupMemberByIDKey(memberID math.Uint) []byte {
 	return append(GroupMemberByIDPrefix, memberID.Bytes()...)
 }
 
+// PolicyPrefixQueue is the canonical key to store policy key.
+//
+// Key format:
+// - <key_prefix><exp_bytes><policy_id_bytes>
+func PolicyPrefixQueue(exp *time.Time, key []byte) []byte {
+	policyByExpTimeKey := PolicyByExpTimeKey(exp)
+	return append(policyByExpTimeKey, key...)
+}
+
+// PolicyByExpTimeKey returns a key with key prefix, expiry
+//
+// Key format:
+// - <key_prefix><exp_bytes>
+func PolicyByExpTimeKey(exp *time.Time) []byte {
+	// no need of appending len(exp_bytes) here, `FormatTimeBytes` gives const length everytime.
+	return append(PolicyQueueKeyPrefix, sdk.FormatTimeBytes(*exp)...)
+}
+
+// GroupMemberPrefixQueue is the canonical key to store group member key.
+//
+// Key format:
+// - <key_prefix><exp_bytes><member_id_bytes>
+func GroupMemberPrefixQueue(exp *time.Time, key []byte) []byte {
+	memberByExpTimeKey := GroupMemberByExpTimeKey(exp)
+	return append(memberByExpTimeKey, key...)
+}
+
+// GroupMemberByExpTimeKey returns a key with key prefix, expiry
+//
+// Key format:
+// - <key_prefix><exp_bytes>
+func GroupMemberByExpTimeKey(exp *time.Time) []byte {
+	// no need of appending len(exp_bytes) here, `FormatTimeBytes` gives const length everytime.
+	return append(GroupMemberQueueKeyPrefix, sdk.FormatTimeBytes(*exp)...)
+}
+
+func ParseIdFromQueueKey(key []byte) math.Uint {
```

Comment:
Please write UTs to fully test these functions.

Reply:
Done